### PR TITLE
Add support for query params in callApi

### LINKED ISSUE #157: Add query string support to the `callApi` function
We'll need to pass query parameters to the `callApi` function soon.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #157

---

This PR adds basic support for query parameters.

### CODE REVIEW

File: src/api/index.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ type CallApiParams = {
   method?: HttpMethod;
   version?: string;
   apiState: ApiState;
+  query?: { [key: string]: string };
```
Comment (bobsilverberg) [inline src/api/index.tsx]: Nit: Do you think maybe `params` would be a better name for this?

Comment (willdurand) [reply src/api/index.tsx]: `params` is a bit too generic to me. `query` could be generic but in this context, it is more meaningful. `queryParameters` would be too long, and `qs` or `queryString` would be a bad name for an object value.

**Comment by willdurand** [comment]
> Thanks for doing this @willdurand. I noticed that we need this for the `browse` API for selecting individual files, and I was going to defer that, but now with this I guess I'll include that in my first iteration.

Yeah, I started to play with that so that clicking a file in the tree list (cf. #153) shows its content on the right panel:

screen shot 2019-02-07 at 11 54 31

That's why I needed this patch :D

**Comment by kumar303** [comment]
Sorry that I didn't have time to review this. I filed a request for changes but it's not high priority: https://github.com/mozilla/addons-code-manager/issues/164

**Pull request merged**